### PR TITLE
This closes kriskowal/q-io#65 (70 tests, 134 assertions, 0 failures, 0 s...

### PR DESCRIPTION
**Pull request by ghost**
copyTree() amended to support pre-existing destination directories - changes on v1 branch. Current unit tests pass.

### CODE REVIEW

**Comment by kriskowal** [comment]
I will land this. I am going to do a code-review, as if I were bringing you on as a long-term contributor. If you do not make these changes, I will myself after I merge. Thanks for putting in the effort.

File: fs-common.js [hunk 1]
```diff
@@ -159,8 +159,8 @@ exports.update = function (exports, workingDirectory) {
             if (stat.isFile()) {
                 return self.copy(source, target);
             } else if (stat.isDirectory()) {
-                return Q.when(self.makeDirectory(target), function () {
-                    return Q.when(self.list(source), function (list) {
```
Comment (kriskowal) [inline fs-common.js:163]: Point of style, in this project (and in the style set out by CommonJS) names are seldom abbreviated. By convention, `target` is the opposite of `source`, so preferred over `destination`. This variable would be called `targetExists`.

File: fs-common.js [hunk 2]
```diff
@@ -168,6 +168,13 @@ exports.update = function (exports, workingDirectory) {
                             );
                         }));
                     });
+                    if (dstExists) {
+                        return copySubTree;
+                    } else {
```
Comment (kriskowal) [inline fs-common.js:173]: At some point, I need to migrate away from `Q.when`. When Q hits 1.0, the supported style will be `Q(variable).then(…)` in the general case, and optimized down to `promise.then(…)` only if the promise is absolutely known to have been created by the `Q` in the same package.

**Comment by kriskowal** [comment]
Looks good to me.

File: fs-common.js (hunk 1 above)
Comment (ghost) [inline fs-common.js:163]: I'll make the change. I appreciate the feedback.

Comment (ghost) [inline fs-common.js:163]: Done - I hope (github is continually auto-logging me out??).

**Pull request merged**